com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ApplicationColorDetector.java

```diff
@@ -100,13 +100,59 @@ private ApplicationColor getOlderApplicationColor(DeployedMta deployedMta) {
                           .orElse(null);
     }
 
-    private Phase getPhaseFromHistoricProcess(String processInstanceId) {
-        HistoricVariableInstance phaseVariableInstance = flowableFacade.getHistoricVariableInstance(processInstanceId, Variables.PHASE.getName());
-        if (phaseVariableInstance == null) {
-            return null;
-        }
+    private Optional<Operation> getLastOperationsWithFinalState(String correlationId) {
+        List<Operation> operations = getOperations(correlationId);
 
-        return Phase.valueOf((String) phaseVariableInstance.getValue());
+        Map<String, List<HistoricOperationEvent>> operationsWithHistoricEvents = operations.stream()
+                                                                                           .filter(operation -> operation.getState() == null)
+                                                                                           .collect(Collectors.toMap(Operation::getProcessId,
+                                                                                                                     this::getOperationHistoricEvents));
+
+        return operations.stream()
+                         .map(operation -> getFinalState(operation, operationsWithHistoricEvents))
+                         .filter(operation -> operation.getState() != null)
+                         .sorted(Comparator.comparing(Operation::getEndedAt)
+                                           .reversed())
+                         .findFirst();
+    }
+
+    private List<Operation> getOperations(String correlationId) {
+        Operation currentOperation = operationService.createQuery()
+                                                     .processId(correlationId)
+                                                     .singleResult();
+
+        return operationService.createQuery()
+                               .mtaId(currentOperation.getMtaId())
+                               .processType(ProcessType.BLUE_GREEN_DEPLOY)
+                               .spaceId(currentOperation.getSpaceId())
+                               .orderByEndTime(OrderDirection.DESCENDING)
+                               .list();
+    }
+
+    private List<HistoricOperationEvent> getOperationHistoricEvents(Operation operation) {
+        return historicOperationEventService.createQuery()
+                                            .processId(operation.getProcessId())
+                                            .list();
+    }
+
+    private Operation getFinalState(Operation operation, Map<String, List<HistoricOperationEvent>> operationsWithHistoricEvents) {
```

Comment:
This can be changed like so:
<pre>
if (operation.getState() == null) {
   return operation;
}
List<> opHistoricEvents = getOperationHistoricEvents(operation);
</pre>
and the rest can be the same